x-pack/plugins/index_management/public/application/components/mappings_editor/__jest__/client_integration/mappings_editor.test.tsx

```diff
@@ -166,12 +205,183 @@ describe('<MappingsEditor />', () => {
         await selectTab('advanced');
       });
 
-      isDynamicMappingsEnabled = find('advancedConfiguration.dynamicMappingsToggle.input').props()[
-        'aria-checked'
-      ];
+      isDynamicMappingsEnabled = getToggleValue(
+        'advancedConfiguration.dynamicMappingsToggle.input'
+      );
       expect(isDynamicMappingsEnabled).toBe(false);
       isNumericDetectionVisible = exists('advancedConfiguration.numericDetection');
       expect(isNumericDetectionVisible).toBe(false);
     });
   });
+
+  describe('component props', () => {
+    /**
+     * Note: the "indexSettings" prop will be tested along with the "analyzer" parameter on a text datatype field,
+     * as it is the only place where it is consumed by the mappings editor.
+     */
+    const defaultMappings: any = {
+      dynamic: true,
+      numeric_detection: false,
+      date_detection: true,
+      properties: {
+        title: { type: 'text' },
+        address: {
+          type: 'object',
+          properties: {
+            street: { type: 'text' },
+            city: { type: 'text' },
+          },
+        },
+      },
+      dynamic_templates: [{ initial: 'value' }],
+      _source: {
+        enabled: true,
+        includes: ['field1', 'field2'],
+        excludes: ['field3'],
+      },
+      _meta: {
+        some: 'metaData',
+      },
+      _routing: {
+        required: false,
+      },
+    };
+
+    let testBed: MappingsEditorTestBed;
+
+    beforeEach(async () => {
+      testBed = await setup({ value: defaultMappings, onChange: onChangeHandler });
+    });
+
+    test('props.value => should prepopulate the editor data', async () => {
+      const {
+        actions: { selectTab, getJsonEditorValue, getComboBoxValue, getToggleValue },
+        find,
+      } = testBed;
+
+      /**
+       * Mapped fields
+       */
+      expect(find('fieldsListItem').length).toEqual(Object.keys(defaultMappings.properties).length);
+
+      const fields = find('fieldsListItem.fieldName').map(item => item.text());
+      expect(fields).toEqual(Object.keys(defaultMappings.properties).sort());
+
+      /**
```

Comment:
This could be something like:

```js
      test('dynamic templates JSON is rendered in the templates editor', async () => {
        await act(async () => {
          await selectTab('templates');
        });

        const templatesValue = getJsonEditorValue('dynamicTemplatesEditor');
        expect(templatesValue).toEqual(defaultMappings.dynamic_templates);
      });
```

Reply:
Added a 1 line comment on top of the existing test